### ISSUE TITLE
[Linux] Stuck on game connecting while trying to watch/start game.

### ISSUE THREAD

**Issue by thorncorona** [ValveSoftware/Dota-2#877]
Setup is listed below. Issue occurs when I try to connect to an online game, and the game gets stuck on the loading screen until SourceTV times out/connection timer ends. Dota works fine in local games however.

Let me know if you need more info. 

```
 OS: Arch Linux 
 Kernel: x86_64 Linux 4.5.4-1-ARCH
 Uptime: 4h 27m
 Packages: 1293
 Shell: bash 4.3.42
 Resolution: 3600x1080
 DE: XFCE4
 WM: Xfwm4
 WM Theme: Numix

 CPU: Intel Core i7-4770K CPU @ 3.9GHz
 GPU: Gallium 0.4 on AMD TAHITI (DRM 2.43.0, LLVM 3.8.0)
 RAM: 15912MiB
```

```
CL:  CLoopModeLevelLoad::OnClientFrameSimulate switching to "game" loopmode with addons: 
SwitchToLoop game requested:  id [1] addons []
Loaded default network config file.  Loaded revision 37 OK
Loaded cached network config file.  Data contains revision 37, not newer than current revision 37; ignoring.
SV:  maxplayers set to 1
Initializing script VM...
...done
Network configuration revision changed 0 -> 37
Got network config from CDN.  Data contains revision 37, not newer than current revision 37; ignoring.
File error loading resource header "panorama/images/iso_countryflags/_png.vtex_c" (Error: ERROR_FILEOPEN)
File error loading resource header "models/controllers/vr_controller.vmdl_c" (Error: ERROR_FILEOPEN)
SV:  Spawn Server: <empty>
Received weekend tourney schedule.
WriteSteamRemoteStorageFileAsync( "cfg/chat.cfg" ) -> at 7.914
Failed to load inventory scratchpad information file "cfg/dota_player_scratchpad.txt"!
CAsyncWriteInProgress::OnComplete( "cfg/chat.cfg" ) -> Success at 7.915
CL:  CWaitForGameServerStartupPrerequisite done waiting for server
CL:  CCreateGameClientJob creating client connection to 'loopback'
CL:  Sending connect to loopback
SV:  Sending S2C_CHALLENGE [2726064356 auth 2] to loopback
CL:  Received S2C_CHALLENGE [2726064356 auth 2] from loopback
CL:  Sending C2S_CONNECT [44 protocol 2726064356 auth 2] to loopback
SV:  Receiving C2S_CONNECT [protocol 44 2726064356 auth 2] from loopback
SV:  Sending S2C_CONNECTION to loopback [addons:'']
SV:  Sending S2C_CONNECTION to loopback [addons:'']
CL:  Received S2C_CONNECTION from loopback [addons:'']
CL:  Connected to 'loopback'
Setting max routable payload size from 1200 to 800 for client
SV:  Sending server info to client 'Goddess <3' at loopback
SV:  WriteInitialSpawnGroups sending 1 groups
Setting max routable payload size from 1200 to 800 for Goddess <3
Game: "Dota 2"
Map: "<empty>"
Players: 1 (0 bots) / 1 humans
Build: 7141 (revision 3458209)
Server Number: 1
CL:  CNetworkGameClient::ProcessServerInfo
Initializing script VM...
...done
ReadSteamRemoteStorageFile( bufOut, "voice_ban.dt" ) -> 0.000180 seconds
CL:  CGameClientConnectPrerequisite connection succeeded
ChangeGameUIState: DOTA_GAME_UI_STATE_LOADING_SCREEN -> DOTA_GAME_UI_STATE_DASHBOARD
SV:  IGameSystem::LoopActivateAllSystems
HO:  IGameSystem::LoopActivateAllSystems
SV:  Game started
CL:  IGameSystem::LoopActivateAllSystems
ReadSteamRemoteStorageFile( bufOut, "scripts/control_groups.txt" ) -> 0.000203 seconds
CL:  CNetworkGameClient::OnSwitchLoopModeFinished( game : success )
SV:  Sending full update to client Goddess <3 (reason:  initial update)
ShouldClientReceiveStringTableUserData called with NULL GameRules() on table userinfo, string 0
CL:  Receiving uncompressed update from server
CL:  Signon traffic "client":  incoming 34.028 KB [6 pkts], outgoing 1.155 KB [7 pkts]
Ping measurement complete.
Failed to communicate with routing cluster 'hkg'.
Communications with routing cluster 'hkg' established.
CL:  CPhysicsGameSystem::UnloadSpawnGroupPhysics(scenes/battlepass_ti6/twoflags) no such group
File error loading resource header "panorama/images/iso_countryflags/_png.vtex_c" (Error: ERROR_FILEOPEN)
StartWatchingGame request SteamID: [A:1:4169432069:6836], watching SteamID: [I:0:0]
Received CMsgWatchGameResponse result 0.
Received CMsgWatchGameResponse result 1.
StartWatchingGame: Sending console command: connect_hltv 192.69.96.185:28023
SwitchToLoop remoteconnect requested:  id [2] addons []
CL:  IGameSystem::LoopDeactivateAllSystems
WriteSteamRemoteStorageFileAsync( "scripts/control_groups.txt" ) -> at 25.660
CL:  Disconnecting from server: #GameUI_Disconnect_LoopDeactivate
CL:  CPhysicsGameSystem::UnloadSpawnGroupPhysics(scenes/shining_default) no such group
CL:  CPhysicsGameSystem::UnloadSpawnGroupPhysics(scenes/home_button) no such group
CL:  CPhysicsGameSystem::UnloadSpawnGroupPhysics(scenes/play_searching) no such group
CL:  CPhysicsGameSystem::UnloadSpawnGroupPhysics(scenes/find_match_status) no such group
CL:  CPhysicsGameSystem::UnloadSpawnGroupPhysics(backgrounds/dashboard_parallax_ti6_initial) no such group
SV:  IGameSystem::LoopDeactivateAllSystems
HO:  IGameSystem::LoopDeactivateAllSystems
SV:  Disconnect client 'Goddess <3' from server(1): NETWORK_DISCONNECT_SHUTDOWN
WriteSteamRemoteStorageFileAsync( "voice_ban.dt" ) -> at 25.796
Network socket 'client' opened on port -1
CL:  CNetworkGameClient::OnSwitchLoopModeFinished( remoteconnect : success )
CAsyncWriteInProgress::OnComplete( "scripts/control_groups.txt" ) -> Success at 25.884
CAsyncWriteInProgress::OnComplete( "voice_ban.dt" ) -> Success at 25.884
CL:  Sending connect to 192.69.96.185:28023
CL:  Received S2C_CHALLENGE [129682995 auth 3] from 192.69.96.185:28023
CL:  Sending C2S_CONNECT [44 protocol 129682995 auth 3] to 192.69.96.185:28023
CL:  Received S2C_CONNECTION from 192.69.96.185:28023 [addons:'']
CL:  Connected to '192.69.96.185:28023'
CL:  Suppress INetchannel::Transmit() in loopmode( remoteconnect )
CL:  CLoopModeRemoteConnect::OnClientFrameSimulate switching to "levelload" loopmode with addons:  
SwitchToLoop levelload requested:  id [2] addons []
ChangeGameUIState: DOTA_GAME_UI_STATE_DASHBOARD -> DOTA_GAME_UI_STATE_LOADING_SCREEN
CL:  CNetworkGameClient::OnSwitchLoopModeFinished( levelload : success )
CL:  Permit INetchannel::Transmit()
Setting max routable payload size from 1200 to 800 for client
CL:  CLoopModeLevelLoad::OnClientFrameSimulate switching to "game" loopmode with addons: 
SwitchToLoop game requested:  id [2] addons []
CL:  CNetworkGameClient already exists for connection to '192.69.96.185:28023'
CL:  connection to '192.69.96.185:28023' already started
```

**Comment by gdrewb-valve** [comment]
Your problem is likely not Linux-specific so you might want to look at the existing threads on this kind of problem at dev.dota2.com.

**Comment by thorncorona** [comment]
Alright I actually managed to get it solved. Woohoo!

Here's how I did it:

After @gdrewb-valve mentioned the fact it may not be a Linux-specific issue, I tried installing Dota 2 through wine. The same symptoms appeared - leading me to believe that the issue was related to my networking setup (has both ethernet and wifi). 

I then tried playing Dota after unplugging my ethernet cable, and voila! It worked. Because of this, I went through my services, and noticed that a systemd service handling my wifi connection was also controlling my ethernet connection (networkmanager). Thus, I disabled my ethernet only connection (for me it was dhcpd@eno1.service) and Dota worked again.
